anndata/tests/test_backed_sparse.py

```diff
@@ -49,17 +87,21 @@ def test_backed_indexing(ondisk_equivalent_adata, subset_func, subset_func2):
         pytest.param(sparse.csc_matrix, sparse.hstack),
     ],
 )
-def test_dataset_append_memory(tmp_path, sparse_format, append_method):
-    h5_path = tmp_path / "test.h5"
+def test_dataset_append_memory(tmp_path, sparse_format, append_method, diskfmt):
+    path = (
+        tmp_path / f"test.{diskfmt.replace('ad', '')}"
+    )  # diskfmt is either h5ad or zarr
```

Comment:
Added this and the one below.  These tests are "positive" in that they tests if appending works but I didn't add `zarr` to the negative tests (wrong shape/format).  I can though, but like I said, we already have slow tests so not sure it's super necessary.